Push/PushConfiguration.swift

```diff
@@ -3,80 +3,67 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
 import Foundation
+import MozillaAppServices
 
-public enum PushConfigurationLabel: String {
-    case fennec = "Fennec"
-    case fennecEnterprise = "FennecEnterprise"
-    case firefoxBeta = "FirefoxBeta"
-    case firefoxNightlyEnterprise = "FirefoxNightly"
-    case firefox = "Firefox"
-
-    public func toConfiguration() -> PushConfiguration {
-        switch self {
-        case .fennec: return FennecPushConfiguration()
-        case .fennecEnterprise: return FennecEnterprisePushConfiguration()
-        case .firefoxBeta: return FirefoxBetaPushConfiguration()
-        case .firefoxNightlyEnterprise: return FirefoxNightlyEnterprisePushConfiguration()
-        case .firefox: return FirefoxPushConfiguration()
-        }
-    }
-}
-
-public protocol PushConfiguration {
-    var label: PushConfigurationLabel { get }
+let PUSH_PROD_HOST = "updates.push.services.mozilla.com"
+let PUSH_STAGE_HOST = "updates-autopush.stage.mozaws.net"
 
-    /// The associated autopush server should speak the protocol documented at
-    /// http://autopush.readthedocs.io/en/latest/http.html#push-service-http-api
-    /// /v1/{type}/{app_id}
-    /// type == apns
-    /// app_id == the “platform” or “channel” of development (e.g. “firefox”, “beta”, “gecko”, etc.)
-    var endpointURL: NSURL { get }
-}
-
-public struct FennecPushConfiguration: PushConfiguration {
-    public init() {}
-    public let label = PushConfigurationLabel.fennec
-    public let endpointURL = NSURL(string: "https://updates.push.services.mozilla.com/v1/apns/fennec")!
-}
-
-public struct FennecStagingPushConfiguration: PushConfiguration {
-    public init() {}
-    public let label = PushConfigurationLabel.fennec
-    public let endpointURL = NSURL(string: "https://updates-autopush.stage.mozaws.net/v1/apns/fennec")!
-}
+class InvalidSchemeError: Error {
```

Comment:
I don't have context overall, but, is there a reason why we're using a class here? Generally, it's more swifty to declare errors as enums and given that you want a string, you'd have an enum with an associated type.

Reply:
This is great feedback! I precisely do need more context on the swifty way of doing things!